contrib/babelfishpg_tds/src/backend/tds/tdsbulkload.c

```diff
@@ -599,34 +651,53 @@ SetBulkLoadRowData(TDSRequestBulkLoad request, StringInfo message)
 					}
 					else
 					{
-						StringInfo plpStr;
-						ParameterToken temp = palloc0(sizeof(ParameterTokenData));
+						ParameterToken token = palloc0(sizeof(ParameterTokenData));
 
-						retStatus = ReadBcpPlp(temp, &message, request);
+						retStatus = ReadBcpPlp(token, &message, request);
 
 						CheckPLPStatusNotOK(request, retStatus, i);
-						if (temp->isNull) /* null */
+						if (token->isNull) /* null */
 						{
 							rowData->isNull[i] = true;
 							i++;
-							temp->isNull = false;
+							token->isNull = false;
 							continue;
 						}
 
-						plpStr = TdsGetPlpStringInfoBufferFromToken(message->data, temp);
-						rowData->columnValues[i] = *plpStr;
-						pfree(plpStr);
-						pfree(temp);
+						temp = TdsGetPlpStringInfoBufferFromToken(message->data, token);
```

Comment:
Shouldn't you use a different pointer for this? Before this line, temp was pointing to another set of StringInfo memory that you allocated at the beginning of the function.

Reply:
I have free'd the temp pointer just before reassigning it